### ISSUE TITLE
Can current token see next token in encoder for code completion ?

### ISSUE THREAD

**Issue by mf1832146** [bayesgroup/code_transformers#5]
Hello, I have not found there exists a mask matrix in encoder for code completion, so every node of ast will participate in self attention computation?

**Comment by jxzhn** [comment]
I found it here (cc/main/src/model.py: line 181)
```python
        w = w * b - INF * (1 - b)
```
where b is from lower triangular matrix `bias` defined on lines 77-79
```python
        self.register_buffer(
            "bias", torch.tril(torch.ones(n_ctx, n_ctx)).view(1, 1, n_ctx, n_ctx)
        )
```

**Comment by mf1832146** [comment]
@jxzhn Thanks, it looks right!!!

**Comment by serjtroshin** [comment]
@jxzhn thanks for helping, I am closing this issue
